lib/smart_proxy_dns_powerdns/dns_powerdns_configuration_validator.rb

```diff
@@ -0,0 +1,36 @@
+require 'smart_proxy_dns_powerdns/dns_powerdns_plugin'
+
+module Proxy::Dns::Powerdns
+  class ConfigurationValidator
+    def validate_settings!
+      backend = settings[:powerdns_backend]
+
+      unless backend
+        settings[:powerdns_backend] = 'mysql' if settings[:powerdns_mysql_hostname]
```

Comment:
I would suggest not to modify plugin settings -- they should be treated as read-only (probably will be enforced in the future). I would also to force users to provide values for critical settings (such as powerdns_backend) instead of deriving them from other values, you are already familiar with 'validate_presence' validator.